src/converters/artifact-converter.ts

```diff
@@ -0,0 +1,91 @@
+import * as aws from '@pulumi/aws';
+import * as cx from 'aws-cdk-lib/cx-api';
+import { getAccountId, getPartition, getRegion } from '@pulumi/aws-native';
+import { FileAssetManifest } from '../assembly';
+import { FileAssetPackaging } from 'aws-cdk-lib/cloud-assembly-schema';
+import { zipDirectory } from '../zip';
+import { StackComponentResource } from '../types';
+
+/**
+ * ArtifactConverter
+ */
+export abstract class ArtifactConverter {
```

Comment:
Nit: does it have go be an abstract class?  resolvePlaceholders could take stackComponent (or pulumi.Resource even) as an extra parameter and this would be just a function. Does it need state in the future?

Reply:
It's an abstract class because we have multiple types of things that will extend it (`FileAssetConverter`, `DockerAssetConverter`, `StackConverter`).